pkgs/development/cuda-modules/fixups/cuda/cuda_nvcc.nix

```diff
@@ -0,0 +1,77 @@
+{
+  lib,
+  backendStdenv,
+  cudaOlder,
+  setupCudaHook,
+}:
+prevAttrs: {
+  # Merge "bin" and "dev" into "out" to avoid circular references
+  outputs = builtins.filter (
+    x:
+    !(builtins.elem x [
+      "dev"
+      "bin"
+    ])
+  ) prevAttrs.outputs or [ ];
+
+  # Patch the nvcc.profile.
+  # Syntax:
+  # - `=` for assignment,
+  # - `?=` for conditional assignment,
+  # - `+=` to "prepend",
+  # - `=+` to "append".
+
+  # Cf. https://web.archive.org/web/20230308044351/https://arcb.csc.ncsu.edu/~mueller/cluster/nvidia/2.0/nvcc_2.0.pdf
+
+  # We set all variables with the lowest priority (=+), but we do force
+  # nvcc to use the fixed backend toolchain. Cf. comments in
+  # backend-stdenv.nix
+
+  postPatch =
+    let
+      nvvmReplace = lib.optionalString (cudaOlder "12.5") ''
+        --replace-fail \
+          '$(TOP)/$(_NVVM_BRANCH_)' \
+          "''${!outputBin}/nvvm" \
+      '';
+    in
+    prevAttrs.postPatch or ""
+    + ''
+      substituteInPlace bin/nvcc.profile \
+        --replace-fail \
+          '$(TOP)/$(_TARGET_DIR_)/include' \
+          "''${!outputDev}/include" \
+          ${nvvmReplace}
+
+      cat << EOF >> bin/nvcc.profile
+
+      # Fix a compatible backend compiler
+      PATH += "${backendStdenv.cc}/bin":
+
+      # Expose the split-out nvvm
+      LIBRARIES =+ "-L''${!outputBin}/nvvm/lib"
+      INCLUDES =+ "-I''${!outputBin}/nvvm/include"
+      EOF
+    '';
+
+  # NOTE(@connorbaker):
+  # Though it might seem odd or counter-intuitive to add the setup hook to `propagatedBuildInputs` instead of
+  # `propagatedNativeBuildInputs`, it is necessary! If you move the setup hook from `propagatedBuildInputs` to
+  # `propagatedNativeBuildInputs`, it stops being propagated to downstream packages during their build because
+  # setup hooks in `propagatedNativeBuildInputs` are not designed to affect the runtime or build environment of
+  # dependencies; they are only meant to affect the build environment of the package that directly includes them.
```

Comment:
I thought there used to be a different message here (talking about offsets), and I'm kind of confused by the new one. "only meant to affect the build environment of the package that directly includes them"?

Reply:
It's the same one: https://github.com/NixOS/nixpkgs/blob/8b19a28ab7a307543536e0253aeacb2941672419/pkgs/development/cuda-modules/cuda/overrides.nix#L252-L257

Replaced in the latest.

Reply:
It's the same one, but I've replaced it with something more succinct in the latest.

Reply:
That one I'm not sure about :(

Reply:
TLDR: this part of the comment is erroneous too, so I'd leave it out while we're at it.

Btw, I think I get it now: `buildInputs` means "same (host) platform as the current package", so relative to us their offsets are `(0, 0)`, and `nativeBuildInputs` are shifted by `(-1, -1)`. My original code was adding `(1, 0)` instead of the current `(0, 0)`